### PR TITLE
Adding ability to create admin users, modified unit tests and added DB scripts

### PR DESCRIPTION
**Pull request by si-x-li**

### CODE REVIEW

File: backend/logic/users/users.js [hunk 1]
```diff
@@ -14,33 +14,141 @@ var insertStudentUser = async (username, password, email, id) => {
...
+    throw new Error("failed to establish connection with database");
+  }
+
+  try {
+    await connection.beginTransaction();
+    let student_id = await connection.query('SELECT student_id FROM students WHERE username = ?;', username)
+    await connection.query('DELETE FROM students WHERE student_id = ?;', [student_id[0].student_id]);
+    await connection.query('DELETE FROM users WHERE username = ?;', [username]);
+    await connection.commit();
+    return true;
```
Comment (zerfuth) [inline backend/logic/users/users.js]: Missing connection.release()?
```suggestion
    connection.release();
    return true;
```

File: backend/logic/users/users.js [hunk 2]
```diff
@@ -14,33 +14,141 @@ var insertStudentUser = async (username, password, email, id) => {
...
+    throw new Error("failed to establish connection with database");
+  }
+
+  try {
+    await connection.beginTransaction();
+    let staff_id = await connection.query('SELECT staff_id FROM staff_members WHERE username = ?;', username)
+    await connection.query('DELETE FROM staff_members WHERE staff_id = ?;', [staff_id[0].staff_id]);
+    await connection.query('DELETE FROM users WHERE username = ?;', [username]);
+    await connection.commit();
+    return true;
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
    connection.release();
    return true;
```

File: backend/logic/users/users.js (hunk 1 above)
Comment (si-x-li) [reply backend/logic/users/users.js]: True good catch

File: backend/routes/users.js [hunk 3]
```diff
@@ -33,17 +33,65 @@ const router = express.Router();
...
 });
 
+/*
+* @api {get} /addStudentUser
+* @apiDescription This endpoint will add a student and an associated user
+* @apiParam (body) {string} username, {string} password, {string} email, {int} admin_id
+* @apiExample {curl} Example usage:
+* Http: 
+	POST /users/addAdminUser HTTP/1.1
+	Host: localhost:3000
```
Comment (KevenLYF) [inline backend/routes/users.js]: new port 3001

**Pull request merged**